pg_search/src/index/reader/index.rs

```diff
@@ -983,6 +923,48 @@ impl SearchIndexReader {
             .expect("search should not fail")
     }
 
+    /// Collect for the given Collector, optionally paired with / wrapped with the given auxiliary
+    /// Collector(s).
```

Comment:
```suggestion
/// Collect for the given Collector, optionally paired with / wrapped with the given auxiliary
/// Collector(s).
///
/// # Type Parameters
/// * `C` - The primary collector (typically a TopDocs collector)
///
/// # Returns
/// A tuple of (primary collector result, optional aggregation results)
```

Reply:
I think that that additional information is obvious from the type signature.